src/functions-and-arrays.js

```diff
@@ -1,109 +1,247 @@
+    return a
+  }else{
+    return b
+  }
 
+}
+
+console.log(maxOfTwoNumbers(5,6))
 
 // Iteration #2: Find longest word
+//                7          7           7             9       5           7        8
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  var longest = 0;
+  var word = null;
+
+  if (words === undefined){
+
+    return null
+    
+  }
 
+  for (var i = 0; i < words.length ; i++) {
+      if (longest < words[i].length) {
+          longest = words[i].length;
+          word = words[i];
+      }
+  }
+  return word;
+}
+console.log(findLongestWord(words))
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+const empty = []
+let numEmpty = 0
+
+if (numbers === undefined){
+
+  return 0
+  
+}
+
+for (i=0; i< numbers.length;i++){
+
+  numEmpty += numbers[i]
+
+}
+  return numEmpty  
+
+
+}
+
+let numEmpty = 0
 
+for (i=0; i< numbers.length;i++){
+
+numEmpty += numbers[i]
+
+}
+console.log(numEmpty)
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum() {
+
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+let numEmpty = 0
+const empty = []
+
+
+if ( numbersAvg.length === 0) {
+  return null
+  
+}
+
+
+for (i=0; i< numbersAvg.length;i++){
+
+  numEmpty += numbersAvg[i]
+
+}
+  return numEmpty / numbersAvg.length
+
+}
+
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) {
+let numEmpty = 0
+
+if (wordsArr.length === 0){
+
+  return null
+  
+}
+
+for (i=0; i< wordsArr.length;i++){
+
+  numEmpty += wordsArr[i].length
+
+}
+  return numEmpty / wordsArr.length
+
+
+}
 
 // Bonus - Iteration #4.1
 function avg() {}
 
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+'crab',
+'poison',
+'contagious',
+'simple',
+'bring',
+'sharp',
+'playground',
+'poison',
+'communion',
+'simple',
+'bring'
 ];
 
-function uniquifyArray() {}
+function uniquifyArray(wordsUnique) {
+let numEmpty = 0
+let nounString = String()
+
+if (wordsUnique.length === 0){
+
+  return null
+  
+}
+
+return wordsUnique.filter((value, index) => wordsUnique.indexOf(value) === index);
 
+}
 
 
 // Iteration #6: Find elements
 const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
 
-function doesWordExist() {}
+function doesWordExist(arr, value) {
+
+if (arr.length === 0 ){
+
+  return null
+  
+}
+
+var status = false;
```

Comment:
var y let tienen diferentes alcances te recomiendo hacer uso de let cuando quieras definir una variable
```suggestion
let status = false;
```